Lib/asyncio/base_events.py

```diff
@@ -630,24 +628,23 @@ def set_default_executor(self, executor):
         self._default_executor = executor
 
     def _getaddrinfo_debug(self, host, port, family, type, proto, flags):
-        msg = ["%s:%r" % (host, port)]
+        msg = [f"{host}:{port!r}"]
         if family:
-            msg.append('family=%r' % family)
+            msg.append(f'family={family!r}' % family)
         if type:
-            msg.append('type=%r' % type)
+            msg.append(f'type={type!r}')
         if proto:
-            msg.append('proto=%r' % proto)
+            msg.append(f'proto={proto!r}')
         if flags:
-            msg.append('flags=%r' % flags)
+            msg.append(f'flags={flags!r}')
         msg = ', '.join(msg)
         logger.debug('Get address info %s', msg)
```

Comment:
F-string here too please 😄

Reply:
No, that's logging :(

`logger.debug('Get address info %s', msg)` -- means that `msg` will be interpolated only if the logging is ON and the logging level is `DEBUG`.  Otherwise, `msg.__str__()` won't be called.  This is a performance thing, as `__repr__()` and `__str__()` can be expensive for some objects.

`logger.debug(f'Get address info {msg}')` would mean that `msg.__str__()` is always called, even when logging is disabled.

Reply:
In this particular case we already computed the `msg`, so we can inline it.  In 99.9% other cases we don't want to convert logging strings.

Reply:
The problem is not only in performance.
Tools like [sentry](https://github.com/getsentry) groups similar log messages basing on the message text. Calls with the same message template but different params are grouped together but pre-formatted templates are not.
As result the tool becames totally unusable: it displays tons of similar but unique messages.